monero/transaction/__init__.py

```diff
@@ -217,13 +258,13 @@ def _scan_pubkeys(svk, psk, stealth_address, amount, encamount):
                 for addridx, addr in enumerate(addresses):
                     psk = binascii.unhexlify(addr.spend_key())
                     payment = _scan_pubkeys(
-                        svk, psk, stealth_address, amount, encamount
+                        svk, psk, stealth_address, amount, encamount, on_chain_vt
                     )
                     if payment:
                         break
             outs.append(
                 Output(
-                    stealth_address=vout["target"]["key"],
+                    stealth_address=stealth_address,
```

Comment:
looks like this is changing the passed in type from a hex string to binary and I would think would cause [these tests](https://github.com/monero-ecosystem/monero-python/blob/d48d615162b0b950549990bf182bcf4fc630c731/tests/test_outputs.py#L64-L83) to fail?

Reply:
Thanks, good spot! hacky'd a fix to set the original type key for both :+1: